### ISSUE TITLE
`Cannot find module` error for Node-provided package

### ISSUE THREAD

**Issue by benmccann** [netlify/zip-it-and-ship-it#743]
**- Do you want to request a _feature_ or report a _bug_?** bug

**- What is the current behavior?** build fails with error `Cannot find module 'stream/package.json'`

**- If the current behavior is a bug, please provide the steps to reproduce.**

- Create a new SvelteKit project with `npm init svelte@next`
- Install the Netlify adapter: https://github.com/sveltejs/kit/tree/master/packages/adapter-netlify#installation
- Create an `.nvmrc` with the contents `16` to use Node 16
- Deploy to Netlify

**- What is the expected behavior?**

There should be no need to install a `stream` package because it is provided by Node: https://nodejs.org/api/stream.html#stream_stream

I'm not sure why it's trying to find `stream/package.json` and failing since this is a built-in package

**- Please mention your node.js, and operating system version.**

Node 16. Linux

**Comment by netlify-team-account-1** [comment]
on it!

**Comment by marovargovcik** [comment]
I face the same issue. I put together a repository where behavior can be reproduced: https://github.com/marovargovcik/netlify-functions-demo Function is located in src/functions directory and is transpilled to ES5 with rollup by running `npm run build` and outputted to dist/functions. Here is the deploy log https://gist.github.com/marovargovcik/ff6989665cc347ce9cf9cba36cbb5598.

**Comment by netlify-team-account-1** [comment]
Did some more digging. This error happens when you're using zip-it-and-ship-it's dependency resolution mechanism, which uses the Node's builtin `process.binding("natives")` to get a list of which dependencies are native (like `fs` or `path` or `stream/web`) and which are not (like `react` or `is-even`). Relying on `process.binding("natives")` means that it depends on whatever Node version is used to do the bundling. Since `stream/web`, which is used by your dependency `fetch-blob`, is a Node v16 feature, it rightfully fails. What zip-it-and-ship-it doesn't detect: `fetch-blob` accounted for that to happen, and has a [fallback strategy](https://github.com/node-fetch/fetch-blob/blob/9022f8fc701306a223510dd86a3f1d56259d8530/streams.cjs#L10). So it fails wrongfully.

You can unblock yourselves by using the `esbuild` bundler (which I see you already found out via https://github.com/sveltejs/kit/pull/2605). In your case @marovargovcik this has the additional upside of replacing the `rollup` bundling stage altogether, since esbuild does all the things rollup does for you.

**Comment by benmccann** [comment]
> it rightfully fails (aws lambda, and by extension netlify functions, don't yet support Node v16)

Ah, this was entirely unclear to me. I had checked [the docs](https://docs.netlify.com/configure-builds/manage-dependencies/) and they make no mention of Node 16 being supported. I had put `16` in my `.nvmrc` file and Netlify didn't seem to complain about it. It would be nice if it just bombed out immediately and told you Node 14 was the latest supported version if you tried to use something newer

**Comment by benmccann** [comment]
`precinct` was upgraded here: https://github.com/netlify/zip-it-and-ship-it/pull/773

However, I'm still getting deployment failures: https://github.com/netlify/cli/pull/3533#issuecomment-958101753

**Comment by benmccann** [comment]
This is now working. Existing projects will need to "clear cache and deploy"

Thanks for all the help!!

**Comment by vldmrkl** [comment]
@benmccann I'm still [experiencing this issue](https://app.netlify.com/sites/modest-tesla-67a211/deploys/61ce756c6f54ff6ff22aaa0b) on my existing project even after clearing the cache and re-deploying.

@netlify-team-account-1 do you know what could go wrong?

I had to change the `node_bundler` to `esbuild` to make it work.

**Comment by netlify-team-account-1** [comment]
Hey @vldmrkl! I'm not aware of any changes that could've lead to this. We introduced a related [test case](https://github.com/netlify/zip-it-and-ship-it/blob/7f2285675f18b065c2da9be5f425e8d059eadc35/tests/main.js#L2351) when this issue first occurred, and that's still green.

Could you open a new issue and provide me with a reproduction case? That'd make it a lot easier for me to help you with this :)
